src/basic/proc-cmdline.c

```diff
@@ -39,6 +39,39 @@ int proc_cmdline(char **ret) {
                 return read_one_line_file("/proc/cmdline", ret);
 }
 
+
+/* note; similar function is also in libshared, but libbasic cant't depend
+* on libshared */
+static bool is_efi_secure_boot() {
+        _cleanup_free_ void *v = NULL;
+        uint8_t b;
+        size_t s;
+        int r;
+
+        r = efi_get_variable(EFI_VENDOR_GLOBAL, "SecureBoot", NULL, &v, &s);
+        if (r < 0)
+                return false;
+
+        if (s != 1)
+                return false;
+
+        b = *(uint8_t *)v;
+        return !!b > 0;
+}
+
+/* In SecureBoot mode this is probably not what you want. As your cmdline is
+ * cryptographically signed like when using Type #2 EFI Unified Kernel Images
+ * (https://systemd.io/BOOT_LOADER_SPECIFICATION/) The user's intention is then
+ * that the cmdline should not be modified.  You want to make sure that the
+ * system starts up as exactly specified in the signed artifact. */
+static int systemd_options_variable(char **line) {
```

Comment:
I wasn't sure what the naming convetion is for "local functions that are slightly different than the function they wrap"